examples/tutorials/subplots.py

```diff
@@ -0,0 +1,239 @@
+# will define our figure to have a 2 row and 3 column grid layout.
+# ``figsize=("15c", "6c")`` defines the overall size of the figure to be 15cm
+# wide by 6cm high. Using ``frame="lrtb"`` allows us to customize the map frame
+# for all subplots instead of setting them individually. The figure layout will
+# look like the following:
+
+with fig.subplot(nrows=2, ncols=3, figsize=("15c", "6c"), frame="lrtb"):
+    for index in range(2 * 3):
+        i = index // 3  # row
+        j = index % 3  # column
+        with fig.set_panel(panel=index):  # sets the current panel
+            fig.text(
+                position="MC",
+                text=f"index: {index}, row: {i}, col: {j}",
+                region=[0, 1, 0, 1],
+            )
+fig.show()
+
+###############################################################################
+# The :meth:`pygmt.Figure.set_panel` command activates a specified subplot, and
+# all subsequent plotting commands will take place in that subplot panel. This
+# is similar to matplotlib's ``plt.sca`` method. In order to specify a subplot,
+# you will need to provide the identifier for that subplot via the ``panel``
+# argument. Pass in either the ``index`` number, or a tuple like
+# (``row``, ``col``) to ``panel``.
+
+###############################################################################
+# .. note::
+#
+#     The row and column numbering starts from 0. So for a subplot layout with
+#     N rows and M columns, row numbers will go from 0 to N-1, and column
+#     numbers will go from 0 to M-1.
+
+###############################################################################
+# For example, to activate the subplot on the top right corner (index: 2) at
+# ``row=0`` and ``col=2``, so that all subsequent plotting commands happen
+# there, you can use the following command:
+
+###############################################################################
+# .. code-block:: default
+#
+#     with fig.set_panel(panel=[0, 2]):
+#         ...
+
+###############################################################################
+# Making your first subplot
+# -------------------------
+# Next, let's use what we learned above to make a 2 row by 2 column subplot
+# figure. We'll also pick up on some new parameters to configure our subplot.
+
+fig = pygmt.Figure()
+with fig.subplot(
+    nrows=2,
+    ncols=2,
+    figsize=("15c", "6c"),
+    autolabel=True,
+    margins=["0.1c", "0.2c"],
+    title="My Subplot Heading",
+):
+    fig.basemap(
+        region=[0, 10, 0, 10], projection="X?", frame=["af", "WSne"], panel=[0, 0]
+    )
+    fig.basemap(
+        region=[0, 20, 0, 10], projection="X?", frame=["af", "WSne"], panel=[0, 1]
+    )
+    fig.basemap(
+        region=[0, 10, 0, 20], projection="X?", frame=["af", "WSne"], panel=[1, 0]
+    )
+    fig.basemap(
+        region=[0, 20, 0, 20], projection="X?", frame=["af", "WSne"], panel=[1, 1]
+    )
+fig.show()
```

Comment:
Some suggestions on this example:

1. this example uses a new way to specify the panel. We should mention that all plotting functions can use the `panel` argument to specify the panel, and subsequent plotting commands without giving `panel` will add more elements to the same panel.
2. The `frame` arguments are the same for all subplots. For this case, we usually can move it to `Figure.subplot()`. Do you think we should do that way instead?

Reply:
Ok, done in f568982d6bcef13f51c125dfd1cf8aeb18360e6f. Added a note for (1) and moved the `frame` to `fig.subplot` for (2).